test/framework/src/main/java/org/elasticsearch/repositories/blobstore/AbstractBlobContainerRetriesTestCase.java

```diff
@@ -289,6 +290,10 @@ protected org.hamcrest.Matcher<Integer> getMaxRetriesMatcher(int maxRetries) {
         return equalTo(maxRetries);
     }
 
+    protected OperationPurpose randomRetryingPurpose() {
+        return randomPurpose();
```

Comment:
Is this on "purpose"?

Reply:
Yes, it's overridden in the S3-specific test suite because that's the only one that won't retry on every purpose.